components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/NativeNotificationBridge.kt

```diff
@@ -37,22 +47,18 @@ internal class NativeNotificationBridge(
         }
 
         with(notification) {
-            loadIcon(iconUrl?.toUri(), origin, Size.DEFAULT)?.let { icon ->
-                builder.setLargeIcon(icon)
+            activityClass?.let {
+                val intent = Intent(context, activityClass)
```

Comment:
Fun Kotlin tip: You can use `.apply { }` to do things on the instance before assignment.

```kotlin
val intent = Intent(context, activityClass).apply {
  putExtra(EXTRA_ON_CLICK, tag)
}
```

What you have here works perfectly well, but I thought I should share. :)

Reply:
I see.  I think I'll have to stick with what it is right now.  The main reason is activityClass could be null and if it is we wouldn't add any pendingIntent to the notification.

Reply:
Thanks for the offline explanation.  I will update this block.